Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -135,14 +136,13 @@ void removeSpacesAround(std::string &text, size_t pos) {
     }
   }
 }
-}  // namespace
-ChemicalReaction *RxnSmartsToChemicalReaction(
-    const std::string &origText,
-    std::map<std::string, std::string> *replacements, bool useSmiles,
-    bool allowCXSMILES) {
+
+std::unique_ptr<ChemicalReaction> parseReaction(
+    const std::string &origText, const ReactionSmartsParserParams &params,
+    bool useSmiles) {
```

Comment:
Any reason useSmiles was not moved the ReactionSmartsParserParams?

Reply:
If you want to create a reaction from SMILES in the v2 API you just call `ReactionFromSmiles()` instead of `ReactionFromSmarts()` this seems a lot simpler and more logical to me